safemysql.class.php

```diff
@@ -170,6 +175,43 @@ public function query()
 		return $this->rawQuery($this->prepareQuery(func_get_args()));
 	}
 
+
+	/**
+	 * Run the provided callback inside a transaction
+	 * @param  callable $callback
+	 * @param  [boolean] $can_retry If the whole transaction should be retried, if it fails due to a deadlock
+	 * @return mixed whatever the callback returns
+	 */
+	public function transaction($callback, $can_retry = false)
```

Comment:
- [ ] Balls. The form of this that I merged is broken. You don't roll back changes made so far in a deadlocked transaction before retrying it.